src/hotspot/share/services/heapDumper.cpp

```diff
@@ -1956,8 +2013,28 @@ class DumpMerger : public StackObj {
     _dump_seq(dump_seq) {}
 
   void do_merge();
+
+  // returns path for the parallel DumpWriter (resource allocated)
+  static char* get_writer_path(const char* base_path, int seq);
+
 };
 
+char* DumpMerger::get_writer_path(const char* base_path, int seq) {
+  // calculate required buffer size
+    size_t buf_size = strlen(base_path)
+                    + 2                 // ".p"
+                    + 1 + (seq / 10)    // number
```

Comment:
Is this suppose to be the number of characters the sequence number will take up in the path? If so, that's not the correct calculation. For example, 35 ends up as 1 + 3 == 4, but it should be 2.

Reply:
I think the easiest solution is probably just coming up with some reasonable max sequence number and then just assuming that you will need enough chars for it. But if you want to actually compute it, there are a few solutions, including a simple div by 10 loop.

Reply:
`sprintf` can be simple to use:
```
 char result[100]; 
 int number_of_digits = sprintf(result, "%d", seq);
```

Reply:
I decided to always reserve 10 symbols (theoretical max number of GC threads is 2^15)